### DISCUSSION TITLE
how to enable csiStorageCapacity ？

### DISCUSSION THREAD

**Discussion by 13567436138** [rook/rook#13970] [Q&A]
how to enable csiStorageCapacity ？

**Comment by Madhu-1** [comment]
You cannot enable it as its not supported.

**Reply by 13567436138** [reply]
not supported，why there is this filed

**Reply by travisn** [reply]
To understand [csiStorageCapacity](https://kubernetes.io/docs/concepts/storage/storage-capacity/), I'm not sure this feature makes sense to enable for the ceph-csi driver as all Ceph storage is available across the full cluster. There is no reason the pod scheduling would be restricted based on storage available since all Ceph storage is available to all nodes. @13567436138 Is there a scenario you see this feature is needed?

**Reply by 13567436138** [reply]
I mean if you don't need it ，why do you invent it？

**Reply by travisn** [reply]
It's a K8s feature, optional for different CSI drivers to implement. It just doesn't apply to Ceph-CSI.

**Reply by Madhu-1** [reply]
Yes what above @travisn mentioned is true, its not applicable to cephcsi and its a kubernetes feature which will be used by other csi drivers.

**Reply by bh-tt** [reply]
Aside from the pod scheduling restriction feature, I think another (perhaps unintended) feature of the capacity is to provide a driver-independent method of showing how much storage is available. If a CSI driver that is available cluster-wide simply creates a single CSIStorageCapacity resource detailing how much storage is still available (in the case of hostbased rook-ceph , by summing up all disks usable for a specific StorageClass), with for example a labelSelector like `kubernetes.io/hostname Exists`, then administrators have an easy overview of how much storage is available for all CSI drivers currently installed.

I'm personally considering a automated PVC expansion operator (which automatically expands the PVC storage requests if the underlying disk nears full). One major consideration was when to stop expanding: you don't want an automated system to completely fill up the available storage, but how would this operator find out the amount of storage still available? I thought using the CSIStorageCapacity would be a nice, driver-independent method of making this work.
